docs/_posts/2020-06-22-on-git.md

```diff
@@ -96,3 +96,46 @@ git push -u origin tile-colors # needed for the first push
 git push # use this for subsequent pushes
 ```
 
+### Merge commits and fast-forwarding
+
+When merging (`git merge`, `git pull`, etc.), depending on your git history, a merge commit is not always created. 
+
+When the git history is linear, you can fast-forward : no merge commit is created.
+
+```
+A - B - C    master         merge 
+         \            --------------->  A - B - C - D  master
+           D feature   (fast-forward)                  feature
+```
+
+When there's a fork, you cannot fast-forward : a merge commit is required.
+
+```
+A - B - C - D master        merge        A - B - C - D - F master
+          \           ----------------->           \   / ^---- merge commit
+            E feature (no fast-forward)              E     feature
+```
+
+By default, `git pull` allows merge commits. To avoid these types of commit, either pull with the flag `--ff-only` (will only merge if it can be fast-forwarded), or `--rebase` (will perform a rebase when needed, do not use if you don't know how to use it). You can use `git config pull.ff only` to automatically use `--ff-only` on this repository.
+
+### Squashing
+
+When merging, if you want to squash all your commits into one, use the flag `--squash`
+
+For example:
+
+```
+A - B              master
+      \                     
+        C - D - E  feature
+```
+
+`git merge --squash feature`
+
+```
+A - B - C' master  (C' = C + D + E)
+```
+
+### Stashing
+
+Sometimes, you do not want to commit your changes, but need to save your changes somewhere (for instance when pulling). You could create a temporary commit, then later rebase and squash, but an easier way is to use the stash. `git stash` puts your unstaged changes into the stash, `git stash drop` applies the latest stashed item and removes it from the stash. Additional commands can be found with `git help stash`.
```

Comment:
`git stash drop` quietly drops the latest entry without applying it. You want `git stash pop`!

Reply:
Fixed in f4803e9.